onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -319,15 +323,18 @@ static void write_scores(std::vector<T>& scores, POST_EVAL_TRANSFORM post_transf
         break;
       case POST_EVAL_TRANSFORM::SOFTMAX:
         ComputeSoftmax(scores);
-        memcpy(Z, scores.data(), scores.size() * sizeof(T));
+        for (auto it = scores.begin(); it != scores.end(); ++it, ++Z)
+          *Z = static_cast<T>(*it);
         break;
```

Comment:
Why not use memcpy? Or std::copy_n?

Or even better, why not write directly to Z in ComputeSoftmax?